bundle/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING 
+      "Choose the type of build, options are: Debug Release RelWithDebInfo." FORCE)
+endif()
+
+if(NOT BUILD_SHARED_LIBS)
+  set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared libraries." FORCE)
+endif()
+
+if(NOT CMAKE_INSTALL_PREFIX)
+  set(CMAKE_INSTALL_PREFIX "${CMAKE_SOURCE_DIR}/install" CACHE STRING
+      "Install path prefix, prepended onto install directories." FORCE)
+endif()
+
+project(serialbox2-all C CXX Fortran)
+enable_testing()
+cmake_minimum_required(VERSION 3.8.1)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake")
+
+include(serialbox2CloneRepository)
+
+set(YODA_GIT_URL "https:///github.com/cosunae/yoda.git" 
+    CACHE PATH "URL of the yoda git repository to clone")
+set(YODA_GIT_BRANCH "create_library" CACHE STRING "Branch of the yoda git repository to clone")
+
+serialbox2_clone_repository(NAME yoda URL ${YODA_GIT_URL} BRANCH ${YODA_GIT_BRANCH} SOURCE_DIR YODA_SOURCE_DIR )
+
+list(APPEND CMAKE_MODULE_PATH "${YODA_SOURCE_DIR}/cmake")
+include(yodaInit)
+
+yoda_init()
+
+# Add custom targets
+yoda_add_target_clean_all(
+  "${CMAKE_BINARY_DIR}/prefix"
+  "${CMAKE_BINARY_DIR}/thirdparty"  
+  "${CMAKE_BINARY_DIR}/yoda-cmake"
+  "${CMAKE_BINARY_DIR}/boost"
+  "${CMAKE_BINARY_DIR}/boost-prefix"
+)
+
+# Set the default download directory (define YODA_DOWNLOAD_DIR)
+yoda_set_download_dir()
+include("thirdparty/Serialbox2AddBoost")
+
+set(serialbox2_cmake_args -DBOOST_ROOT=${BOOST_ROOT})
+set(serialbox2_dependencies "boost")
+
+yoda_find_package(

```

Comment:
This will add Serialbox (itself) as a dependency, and therefore continue processing its CMake and compiling it after all other dependencies are resolved:
* `DEPENDS` will depend all other packages Serialbox depends on, in this case Boost. That means you can compile this bundle in parallel, since there are targets for each of the packages that are declared with the corresponding dependencies. 
* `FORWARD_VARS` declares variables that are forwarded from the dependency package (i.e. Serialbox in this case) to this CMake (i.e. the bundle)